base/src/main/java/uk/ac/ebi/atlas/solr/analytics/fullanalytics/AnalyticsQueryBuilder.java

```diff
@@ -0,0 +1,104 @@
+package uk.ac.ebi.atlas.solr.analytics.fullanalytics;
+
+import com.google.common.collect.Sets;
+import org.apache.solr.client.solrj.SolrQuery;
+import org.apache.solr.client.solrj.util.ClientUtils;
+
+import java.text.MessageFormat;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class AnalyticsQueryBuilder {
+    enum Field {
+        EXPERIMENT_ACCESSION("experiment_accession"),
+        BIOENTITY_IDENTIFIER("bioentity_identifier"),
+        ASSAY_GROUP_ID("assay_group_id"),
+        EXPRESSION_LEVEL("expression_level");
+        final String name;
+
+        Field(String name){
+            this.name = name;
+        }
+
+        @Override
+        public String toString() {
+            return this.name;
+        }
+    }
+
+    private static final String SIMPLE_QUERY_TEMPLATE = "{0}:\"{1}\"";
+    private static final String TERMS_QUERY_TEMPLATE = "('{'!terms f={0}'}'{1})";
+    private static final String RANGE_QUERY_TEMPLATE = "{0}:[{1} TO *]";
```

Comment:
You're using the word "query" to mean the result of this class - an analytics query - and also individual bits here that the query later consists of. Rename these as "clauses", like in SQL?

Reply:
Or: "filters"?

Reply:
Clause seems better, filter can be mis-leading, being this a Solr-related class.

Reply:
After reading Solr’s reference guide [the most widespread term in Solr parlance seems to be “term”](https://lucene.apache.org/solr/guide/6_6/the-standard-query-parser.html#TheStandardQueryParser-SpecifyingTermsfortheStandardQueryParser). I’ll go with that.